RomanGupalo/golang_bootcamp/server/handlers/requests_test.go

```diff
@@ -0,0 +1,384 @@
+
+	// Check row from db
+	db_book, err := entity.GetBook(good_id, connection.GetSession())
+	require.NoError(t, err, err)
+
+	expected := *book.Price
+	actual := *db_book.Price
+	require.Equal(t, expected, actual,
+		"database returned bad record: \ngot: %v\n want: %v",
+		actual, expected)
+}
+
+func TestUpdateBookHandlerOnBadRequest(t *testing.T) {
+	clearDatabase()
+	book, err := createTestBook()
+	require.NoError(t, err, err)
+	str_id := strconv.FormatInt(int64(book.Id), 10)
+
+	// test function to avoid code duplication
+	test_func := func() {
+		rr, err := serveHTTP("PUT", "/books/"+str_id, getBookReader(book), json_content)
+		require.NoError(t, err, err)
+
+		require.Equal(t, rr.Code, http.StatusBadRequest,
+			"handler returned wrong status code: got %v want %v", rr.Code, http.StatusOK)
+	}
+
+	book.Price = nil
+	test_func()
+
+	book.Price = new(float32)
+	*book.Price = -1
+	test_func()
+
+	*book.Price = 0
+	*book.Amount = -1
+	test_func()
+}
+
+func TestDeleteBookHandlerOnSuccess(t *testing.T) {
+	clearDatabase()
+	book, err := createTestBook()
+	require.NoError(t, err, err)
+	str_id := strconv.FormatInt(int64(book.Id), 10)
+
+	rr, err := serveHTTP("DELETE", "/books/"+str_id, nil, text_plain)
+	require.NoError(t, err, err)
+
+	require.Equal(t, rr.Code, http.StatusNoContent,
+		"handler returned wrong status code: got %v want %v", rr.Code, http.StatusNoContent)
+}
+
+func TestDeleteBookHandlerOnNotFound(t *testing.T) {
+	clearDatabase()
+	book, err := createTestBook()
+	require.NoError(t, err, err)
+	str_id := strconv.FormatInt(int64(book.Id), 10)
+	book.Delete(connection.GetSession())
+
+	rr, err := serveHTTP("DELETE", "/books/"+str_id, nil, text_plain)
+	require.NoError(t, err, err)
+
+	require.Equal(t, rr.Code, http.StatusNotFound,
+		"handler returned wrong status code: got %v want %v", rr.Code, http.StatusNotFound)
+}
+
+func makeBook(name string, price float32, genre int, amount int) *entity.Book {
+	book := entity.Book{
+		Name:   &name,
+		Price:  &price,
+		Genre:  &genre,
+		Amount: &amount,
+	}
+	return &book
+}
+func TestGetBooksByFilterHandler(t *testing.T) {
+	clearDatabase()
+	// creating test data set of 100 objects
+	const capacity = 100
+	test_books := make([]*entity.Book, 0, capacity)
+	for i := 0; i < capacity; i++ {
+		name := "only_test_book_" + strconv.FormatInt(int64(i), 10)
+		price := rand.Float32() * 100
+		genre := rand.Uint32()%3 + 1
+		amount := rand.Uint32() % 2
+		buff := makeBook(name, price, int(genre), int(amount))
+		test_books = append(test_books, buff)
+		(*buff).Insert(connection.GetSession())
+	}
+
+	// clearing after testing
+	defer func() {
+		for i := range test_books {
+			(test_books[i]).Delete(connection.GetSession())
+		}
+	}()
+
+	// create map that consist of the book names & expected status
+	// expected status = true -> book must be in returning test set
+	get_expected := func(name string, lp *float32, rp *float32, genre *int) (result map[string]bool) {
```

Comment:
result return field name is redundant, can be ommited:
```suggestion
	getExpected := func(name string, lp *float32, rp *float32, genre *int) map[string]bool {
```